client/src/pages/Shop.js

```diff
@@ -0,0 +1,230 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import Button from "@material-ui/core/Button";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import Paper from "@material-ui/core/Paper";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import CardMedia from "@material-ui/core/CardMedia";
+import Container from "@material-ui/core/Container";
+import SettingsIcon from "@material-ui/icons/Settings";
+import FavoriteBorderOutlinedIcon from "@material-ui/icons/FavoriteBorderOutlined";
+import FavoriteIcon from "@material-ui/icons/Favorite";
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(theme => ({
+  image: {
+    backgroundRepeat: "no-repeat",
+    backgroundSize: "cover",
+    backgroundPosition: "center"
+  },
+  paper: {
+    margin: theme.spacing(8, 4),
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+    height: "100vh",
+    justifyContent: "space-around"
+  },
+  editCoverBtn: {
+    margin: theme.spacing(2)
+  },
+  cardGrid: {
+    paddingTop: theme.spacing(8),
+    paddingBottom: theme.spacing(8)
+  },
+  card: {
+    height: "100%",
+    display: "flex",
+    flexDirection: "column"
+  },
+  cardMedia: {
+    paddingTop: "56.25%" // 16:9
+  },
+  cardContent: {
+    flexGrow: 1,
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    backgroundColor: "black",
+    color: "white"
+  },
+  cardSettingBtn: {
+    position: "relative",
+    left: "100%",
+    transform: "translate(-100%,0)",
+    "&:hover": {
+      color: "grey"
+    }
+  },
+  cardLikedBtn: {
+    position: "relative",
+    transform: "translate(-100%,0)",
+    "&:hover": {
+      color: "grey"
+    }
+  },
+  cardHeader: {
+    height: "0px",
+    width: "100%"
+  }
+}));
+
+export default function Shop() {
+  // material ui styles init
+  const classes = useStyles();
+
+  const [shop, setShop] = useState({
+    cover_photo: "",
+    description: "",
+    items: [{ title: "", photos: [""], price: "" }],
+    title: "",
+    user: ""
+  });
+  const [isMyShop, setIsMyShop] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  const [cardLiked, setCardLiked] = useState(
+    new Array(shop.items.length).fill(false)
+  );
+  // fetch shop data
+  const token = localStorage.getItem("token");
+  const USER_API = "http://localhost:3001/users/";
+  // user id will be from userContext in the future
+  const userid = "5da4c98f912176179cb35e43";
```

Comment:
you can use the _id from the token. Use jwt.decode(token) to get the user details.